arduino/src/main.cpp

```diff
@@ -20,40 +24,111 @@ void setup(){
     while(WiFi.status() != WL_CONNECTED){
         Serial.print(".");
         delay(100);
+        
     }
-
-    
+    pinMode(LED,OUTPUT);
+    pinMode(BUTTON,INPUT_PULLUP);
 }
-
 void loop() {
-    // Odczytaj stan GPIO
-   int gpioState = 10 ;
-    DynamicJsonDocument jsonDoc(200);
-    jsonDoc["gpioState"] = gpioState;
-    String payload;
-    serializeJson(jsonDoc,payload);
-    // Wyślij żądanie POST na serwer Flask
-    HTTPClient http;
-    http.begin(serverAddress);
-    http.addHeader("Content-Type", "application/json");
-     int httpResponseCode = http.sendRequest("GET", payload);
-    if(httpResponseCode > 0)
+    
+    
+    int buttonState = digitalRead(BUTTON);
+   
+    if(buttonState==LOW)
     {
-        Serial.printf("HTTP Response code: %d\n",httpResponseCode );
-        String response = http.getString();
+        
+    // Odczytaj stan GPIO
+        const char* miejsce="warsztat 027";
         DynamicJsonDocument jsonDoc(200);
-        deserializeJson(jsonDoc, response);
-        String success = jsonDoc["success"].as<String>();
-        Serial.println(response);
-
+        jsonDoc["place"] =miejsce ;
+        jsonDoc["state"]="hanged";
+        jsonDoc["board_id"] = BOARD_ID;
+        String payload;
+        serializeJson(jsonDoc,payload);
+        // Wyślij żądanie POST na serwer Flask
+        HTTPClient http;
+        http.begin(SERVER_ADDRESS);
+        http.addHeader("Content-Type", "application/json");

```

Comment:
Ta część kodu się powtarza (czyli budowanie jsona oraz wysyłanie go).

zapakuj to w jakąś funkcję o ładnej nazwie typu `int send_status_request(type, state);